core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlResponseParserUtils.java

```diff
@@ -63,6 +67,24 @@ public static XmlElement parse(SdkPojo sdkPojo, SdkHttpFullResponse response) {
         }
     }
 
+    /**
+     * Gets the Member which is a Payload and which is of Blob Type.
+     * @param sdkPojo
+     * @return Optional of SdkField member if member is Blob type payload else returns Empty.
+     */
+    public static Optional<SdkField<?>> getBlobTypePayloadMemberToUnmarshal(SdkPojo sdkPojo) {
+        return sdkPojo.sdkFields().stream()
+                .filter(f -> f.location() == MarshallLocation.PAYLOAD
+                        || MarshallerUtil.locationInUri(f.location()))
```

Comment:
Why would a payload member be in the URI?

Reply:
Settings that don't make sense for the given input/output direction should default to body. E.g. URI fields in a response should go into the payload.
https://github.com/aws/aws-sdk-java-v2/pull/1996

Reply:
Would it still be an explicit payload member in this case? (Line 79, or would that filter it out?)

Reply:
From Original [#1996_Snippet](https://github.com/aws/aws-sdk-java-v2/commit/97b6ddd2901c59f3ad9006325a94499bcf9bcbfe#diff-ddf3c8c407420734b27fa9e5993efcc25f08b4c47705b9ed3e566d7df5ce9d78R167-R172) which added this for Json ,it looks like it would still be explicit payload member.

Reply:
Can we add a protocol test for this?

Reply:
Thanks Matt , the MarshallerUtil.locationInUri(f.location()) is not necessary in this case. It is a dead code since it will be filtered at next stage where it checks for  f.containsTrait(PayloadTrait.class);

I had taken wrong reference for parsing SDK_Bytes actually it should have been [https://github.com/aws/aws-sdk-java-v2/blob/master/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java#L207-L209](https://github.com/aws/aws-sdk-java-v2/blob/master/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java#L207-L209)
Sorry for the confusion , removed these extra checks.